pkg/api/handlers/compat/images_build.go

```diff
@@ -220,9 +220,11 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 			var m = []string{}
 			if err := json.Unmarshal([]byte(query.Dockerfile), &m); err != nil {
 				// it's not json, assume just a string
-				m = []string{filepath.Join(contextDirectory, query.Dockerfile)}
+				m = []string{query.Dockerfile}
+			}
+			for _, containerfile := range m {
+				containerFiles = append(containerFiles, filepath.Join(contextDirectory, filepath.Clean(filepath.FromSlash(containerfile))))
```

Comment:
is the contextDirectory assured to be safe at this point?  i.e., no way for a symbolic link to get sluffed in?

Reply:
We made it ourselves via extracting a tar file so it should be safe.